3-WebApp-multi-APIs/Controllers/HomeController.cs

```diff
@@ -74,21 +74,26 @@ public async Task<IActionResult> Tenants()
 
         public async Task<IActionResult> Blob()
         {
-            var scopes = new string[] { "https://storage.azure.com/user_impersonation" };
-
-            var accessToken =
-                await tokenAcquisition.GetAccessTokenForUserAsync(scopes);
-
-            // create a blob on behalf of the user
-            TokenCredential tokenCredential = new TokenCredential(accessToken);
-            StorageCredentials storageCredentials = new StorageCredentials(tokenCredential);
-
+            string message = "Blob failed to create";
             // replace the URL below with your storage account URL
             Uri blobUri = new Uri("https://blobstorageazuread.blob.core.windows.net/sample-container/Blob1.txt");
-            CloudBlockBlob blob = new CloudBlockBlob(blobUri, storageCredentials);
-            await blob.UploadTextAsync("Blob created by Azure AD authenticated user.");
-
-            ViewData["Message"] = "Blob successfully created";
+            BlobClient blobClient = new BlobClient(blobUri, new TokenAcquisitionTokenCredential(tokenAcquisition));
+
+            string blobContents = "Blob created by Azure AD authenticated user.";
+            byte[] byteArray = Encoding.ASCII.GetBytes(blobContents);
+            using (MemoryStream stream = new MemoryStream(byteArray))
+            {
+                try
+                {
+                    await blobClient.UploadAsync(stream);
+                    message = "Blob successfully created";
+                }
+                catch (Exception)
+                {
```

Comment:
If you sign-in, restart the application, and go to Blog page again, `MsalUiRequiredException` is thrown and the attribute on top of the ActionResult tries to handle it. For that, you cannot swallow the exception, so I recommend adding this:

```c#
catch(MsalUIRequiredException ex)
{
   throw ex;
}
```

Reply:
Updated.